hardware_applications/week2/final_helpers.cpp

```diff
@@ -0,0 +1,52 @@
+#include <iostream>
+#include "final_helpers.h"
+
+using namespace std;
+
+void executeCommandString(RJRobot &robot, const std::string &commands) {
+    size_t currentPosition = 0;
+    auto nextSpacePosition = commands.find(' ', currentPosition);
+    do {
+        executeOneCommand(robot, commands.substr(currentPosition, nextSpacePosition));
+        currentPosition = nextSpacePosition + 1;
+        nextSpacePosition = commands.find(' ', currentPosition);
+    } while(nextSpacePosition != std::string::npos);
+}
+
+void executeOneCommand(RJRobot &robot, const std::string &command) {
+
+    if(command == "forward") {
```

Comment:
This might be a good place to use cases instead of a ton of if/elseif?

Reply:
It might... except...
https://stackoverflow.com/questions/650162/why-switch-statement-cannot-be-applied-on-strings
TL;DR you can't switch on a string